### PR TITLE
ING-597: Add create agent timeout for buckets tracking manager

### PR DESCRIPTION
**Pull request by chvck**
Motivation
----------
The buckets tracking manager does not deadline new agent creation, this could mean that GetAgent could block indefinitely or our watcher thread could also become blocked - which in turn could also block GetAgent.

Changes
-------
Added a CreateAgentTimeout to buckets tracking manager options. Use the new timeout when creating a context to pass into CreateAgent.

### CODE REVIEW

**Comment by chvck** [comment]
We should probably think about deadlining the manual update http request too.

File: bucketstracking_agentmanager.go [hunk 1]
```diff
@@ -91,12 +91,21 @@ func CreateBucketsTrackingAgentManager(ctx context.Context, opts BucketsTracking
 		zap.Any("bootstrapConfig", bootstrapConfig),
 		zap.String("networkType", networkType))
 
+	createAgentTimeout := opts.CreateAgentTimeout
+	if createAgentTimeout == 0 {
+		createAgentTimeout = 7 * time.Second
```
Comment (Westwooo) [inline bucketstracking_agentmanager.go:96]: Why 7 seconds?

Comment (chvck) [reply bucketstracking_agentmanager.go:96]: 7 seconds is what we typically use as the connection timeout in the SDK, I'm unsure where that value originated.

File: streamwatcher_http.go [hunk 2]
```diff
@@ -139,13 +145,12 @@ func (w *StreamWatcherHttp[StreamT]) watch(
 	close(outCh)
 }
 
-func (w *StreamWatcherHttp[StreamT]) Reconfigure(Buckets *StreamWatcherHttpConfig) error {
+func (w *StreamWatcherHttp[StreamT]) Reconfigure(cfg *StreamWatcherHttpReconfigureConfig) error {
 	w.lock.Lock()
 	w.state = &streamWatcherHttpState{
-		httpRoundTripper: Buckets.HttpRoundTripper,
-		endpoints:        Buckets.Endpoints,
-		userAgent:        Buckets.UserAgent,
```
Comment (Westwooo) [inline streamwatcher_http.go:147]: Why are we removing UserAgent from here?

**Review by Westwooo — COMMENTED** [review]
lgtm, got a couple of questions

File: bucketswatcher_http.go [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+
+type BucketsWatcherHttp struct {
+	stateLock sync.Mutex
+	state     *bucketsWatcherHttpState
+	logger    *zap.Logger
+	userAgent string
+	closed    bool
+
+	fastMap   AtomicPointer[map[string]*Agent]
+	slowMap   map[string]*Agent
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply bucketswatcher_http.go:51]: Can a fast map be used without a lock, whereas a slow map cannot?

File: streamwatcher_http.go (hunk 2 above)
Comment (chvck) [reply streamwatcher_http.go:147]: useragent shouldn't be a part of reconfiguration.

**Review by Westwooo — APPROVED** [review]

**Pull request merged**